process/block/preprocess/smartContractResults.go

```diff
@@ -198,6 +204,8 @@ func (scr *smartContractResults) ProcessBlockTransactions(body block.Body, round
 			continue
 		}
 
+		//TODO: Should be checked max gas limit per block also when a node processes smart contract result transactions from a received block?
```

Comment:
yes. we should for everything. but us an interface, it is not safe to send an pointer to soo many functions.